### PR TITLE
Add connectivity banner to the bottom navigation menu

### PR DESCRIPTION
**Pull request by lazarinibruno**
# Add connectivity banner to the bottom navigation menu

## Description
This PR introduces a banner underneath the bottom app bar which indicates if the user is offline. For this, it uses the `NetworkChangeListener` class that I implemented in #203. 

It closes issue #347.

## Changes

At a first glance, it may seem that there are a lot of changes. However, the only relevant changes are in the files
- `com/android/periodpals/ui/navigation/BottomNavigationMenu.kt`.
- `com/android/periodpals/ui/navigation/BottomNavigationMenuTest.kt`

I had to modify a lot of files since every major screen needs the bottom navigation menu, and the latter needs the `NetworkChangeListener` class.

### Added
None.

### Modified
See files section.

### Removed
None.

## Dependencies Added
None.

## Testing
- I adapted the already existing tests.
- I added some tests to the `BottomNavigationMenuTest` that verify that the bar shows only when the app looses connection.

## Screenshots
image

### CODE REVIEW

**Review by coaguila — APPROVED** [review]
LGTM

File: app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt [hunk 1]
```diff
@@ -12,53 +16,85 @@ import androidx.compose.material3.NavigationBarItem
...
+                        unselectedIconColor = MaterialTheme.colorScheme.onSurface,
+                        unselectedTextColor = MaterialTheme.colorScheme.onSurface,
+                    ),
+            )
+          }
+        },
+    )
+
+    // Offline bar
+    if (!isOnline) {
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt:82]: Nice composable, I really like how it integrates to the screen!

File: app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt [hunk 2]
```diff
@@ -12,53 +16,85 @@ import androidx.compose.material3.NavigationBarItem
...
+      Box(
+          modifier =
+              Modifier.fillMaxWidth()
+                  .height(MaterialTheme.dimens.small3)
+                  .background(color = MaterialTheme.colorScheme.error)
+                  .testTag(CONNECTIVITY_BANNER),
+          contentAlignment = Alignment.TopCenter,
+      ) {
+        Text(
+            text = "No connection",
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt:92]: This should be extracted to the `strings.xml` file

File: app/src/test/java/com/android/periodpals/ui/navigation/BottomNavigationMenuTest.kt [hunk 3]
```diff
@@ -123,4 +137,32 @@ class BottomNavigationMenuTest {
         .onNodeWithTag(BottomNavigationMenu.BOTTOM_NAVIGATION_MENU_ITEM + "Alert List")
         .assertIsSelected()
   }
+
+  @Test
+  fun onlineAppShowsNoBanner() {
```
Comment (charliemangano) [inline app/src/test/java/com/android/periodpals/ui/navigation/BottomNavigationMenuTest.kt:142]: Complete tests, good job!

**Review by charliemangano — APPROVED** [review]
No notes, really good implementation, and it looks great!

File: app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt (hunk 2 above)
Comment (lazarinibruno) [reply app/src/main/java/com/android/periodpals/ui/navigation/BottomNavigationMenu.kt:92]: Its trueeee

**Pull request merged**